src/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -137,12 +137,13 @@ internal async Task<T> ExecuteRequestAsync<T>(Uri endPoint, HttpMethod method, R
                     {
                         httpEvent.OauthErrorCode = JsonHelper.DeserializeFromJson<MsalTokenResponse>(response.Body).Error;
                     }
-                    catch (SerializationException) // in the rare case we get an error response we cannot deserialize
+                    catch (SerializationException e) // in the rare case we get an error response we cannot deserialize
                     {
                         throw MsalExceptionFactory.GetServiceException(
```

Comment:
>MsalExceptionFactory [](start = 30, length = 20)

not this pr. is this really a service exception? It seems to be a client side exception?

Reply:
Debatable. We are expecting the body of the HTTP Message to be a JSON of a certain format. If the service breaks this contract with us, this exception is thrown.